Conference results

rebeccamccabe opened the pull request.

rebeccamccabe committed:
Initial CI attempt with Julia and Gurobi

rebeccamccabe committed:
Merge remote-tracking branch 'origin/main' into cem-ci

rebeccamccabe committed:
Set DVC remote

rebeccamccabe committed:
store notebook html in git to avoid calkit error

rebeccamccabe committed:
use conda env for ci bash commands

rebeccamccabe committed:
pass .hscfg to hsconfigure

rebeccamccabe committed:
fix mkdir of settings folder

rebeccamccabe committed:
run calkit pipeline

rebeccamccabe committed:
fix hsconfigure again

rebeccamccabe committed:
change python version to 3.10

rebeccamccabe committed:
add env vars to calkit pipeline

rebeccamccabe committed:
create calkit run action

rebeccamccabe committed:
make ~/.hscfg directly without using hsconfigure

rebeccamccabe committed:
remove setup python in calkit action

rebeccamccabe committed:
run CI with bash -i instead of bash -l

rebeccamccabe committed:
add conda init

rebeccamccabe committed:
switch name of anaconda environment

rebeccamccabe committed:
fix dependencies in yaml to have proper nesting

rebeccamccabe committed:
comment out conda env activation in run all

rebeccamccabe committed:
run all is run with bash -l again not -i

rebeccamccabe committed:
add more printout for gurobi license setup

rebeccamccabe committed:
change gurobi setup script filepath

rebeccamccabe committed:
let devcontainer see whole project so it can access gurobi script

rebeccamccabe committed:
undo accidental change to hsds config in CI

rebeccamccabe committed:
remove printout from license check since it is called multiple times and could be misleading

rebeccamccabe committed:
fix gurobi dockerfile env vars + executable

rebeccamccabe committed:
remove redundant gurobi test

rebeccamccabe committed:
no verbose wget and make gdown script stop on error

rebeccamccabe committed:
comment gurobi dependency since it needs to be in docker not in overall env

rebeccamccabe committed:
actually comment gurobi dependency since it needs to be in docker not in overall env

rebeccamccabe committed:
add hsds config to calkit

rebeccamccabe committed:
use envsubst instead of cat for hscfg

rebeccamccabe committed:
save dvc cache even if calkit run fails

rebeccamccabe committed:
move shell scripts/commands from dvc to calkit yaml, and change decider to decider-2 for consistency with conda file

rebeccamccabe committed:
comment immediate error and unnecessary pg data download

rebeccamccabe committed:
Dockerfile: add julia and make single stage

rebeccamccabe committed:
try gdown in python instead of cmd line

rebeccamccabe committed:
fix julia stages in calkit

rebeccamccabe committed:
run pipeline

rebeccamccabe committed:
run calkit verbose

rebeccamccabe committed:
delete blank files moved elsewhere

rebeccamccabe committed:
try always committing so env lock file is saved even when run fails

rebeccamccabe committed:
Merge remote-tracking branch 'origin/main' into cem-ci

rebeccamccabe committed:
Update calkit save with fix from https://github.com/calkit/calkit/issues/451

petebachant commented on calkit.yaml:
```diff
@@ -12,14 +12,52 @@ notebooks:
+    setup-hsds:
+      kind: shell-command
+      command: envsubst < .hscfg > ~/.hscfg
+      environment: decider-2
```
It is possible, though maybe risky, to run commands out in the system environment:

```suggestion
      environment: _system
```

rebeccamccabe commented:
HSDS is only used by the python scripts (specifically the setup_sweep stage
notebook) so I think it’s ok for this to remain in the decider-2 conda
environment.

-Becca


On Tue, Aug 5, 2025 at 4:12 PM Pete Bachant ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In calkit.yaml
> <https://github.com/symbiotic-engineering/WEC-DECIDER/pull/8#discussion_r2255243234>
> :
>
> >  pipeline:
>    stages:
> +    setup-hsds:
> +      kind: shell-command
> +      command: envsubst < .hscfg > ~/.hscfg
> +      environment: decider-2
>
> It is possible, though maybe risky, to run commands out in the system
> environment:
> ⬇️ Suggested change
>
> -      environment: decider-2
> +      environment: _system
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/symbiotic-engineering/WEC-DECIDER/pull/8#pullrequestreview-3089583286>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHUJPSXRK4HXJQHUWSQIO6L3MEF3NAVCNFSM6AAAAACC2WI456VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTAOBZGU4DGMRYGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

rebeccamccabe committed:
use real hydro formula for B_u and add D_f as a sweep variable

rebeccamccabe committed:
use * for calkit case dependencies

rebeccamccabe committed:
calkit glob doesnt work, so paste in all case folders

rebeccamccabe committed:
make julia not run if results already exists for that case

rebeccamccabe committed:
refactor results notebook into functions

rebeccamccabe committed:
add forgotten end in julia run script

rebeccamccabe committed:
fix double settings folder in powergenome bash

rebeccamccabe committed:
analyze results working for multidimensional sweep

rebeccamccabe committed:
Run full sweep but get zero wave capacity in results

rebeccamccabe committed:
Merge branch 'cem-ci' of github.com:symbiotic-engineering/WEC-DECIDER into cem-ci

rebeccamccabe committed:
fix missing and duplicate calkit case files

rebeccamccabe committed:
adjust a few more case folders to calkit.yaml

rebeccamccabe committed:
change price from 1000 3000 5000 to 500 700 900

rebeccamccabe committed:
correct dimensionalization error on B_0_e that made CFs too tiny

rebeccamccabe committed:
run sweep again with fixed variability, but still zeros for results

rebeccamccabe committed:
use only inputs from scenario input csv, not all in folder

rebeccamccabe committed:
avoid error for scalar inputs in plot

rebeccamccabe committed:
run more sims and get zeros

rebeccamccabe committed:
finally nonzero wec cap values when using costs 300, 400, 500

rebeccamccabe committed:
stop tracking .calkit/notebooks/executed/modules/CEM/make_additional_tech_csv.ipynb

rebeccamccabe committed:
Runs for conference presentation

rebeccamccabe committed:
Avoid using from-stage-outputs for genx because of parameter dependency crossover

rebeccamccabe commented on modules/CEM/run_all.sh:
```diff
@@ -7,15 +7,16 @@ function run_notebook() {
 
 echo "$(date +"%Y-%m-%d %H:%M:%S") Downloading PowerGenome data"
-bash modules/CEM/get-powergenome-data.sh
+python modules/CEM/get-powergenome-data.py
```
change back to bash

rebeccamccabe committed:
Add modules/CEM/figs

rebeccamccabe committed:
Add modules/CEM/pubs

rebeccamccabe committed:
use bash not python gdown script

rebeccamccabe committed:
Update README with h5pyd configuration steps

Added instructions for configuring h5pyd with conda activation and NREL code.

rebeccamccabe committed:
Update Julia version recommendation in README

Updated Julia version recommendation from 1.8 to 1.8 or 1.9 in the installation instructions.

rebeccamccabe committed:
Update README with PowerGenome metadata manual rename instructions

Added notes on PowerGenome resource groups and metadata fixes.

rebeccamccabe committed:
Merge remote-tracking branch 'origin/continuous-integration' into cem-ci

rebeccamccabe committed:
stop tracking .calkit/notebooks/executed/modules/CEM/make_additional_tech_csv.ipynb

MadisonDietrich committed:
timeseries interpolation

MadisonDietrich committed:
Merge branch 'cem-ci' of https://github.com/symbiotic-engineering/WEC-DECIDER into cem-ci

MadisonDietrich committed:
convert timeseries to sea state matrix

MadisonDietrich committed:
emissions.csv

rebeccamccabe committed:
add instructions for locally modifying GenX to readme

rebeccamccabe committed:
Merge branch 'cem-ci' of github.com:symbiotic-engineering/WEC-DECIDER into cem-ci

rebeccamccabe committed:
turn off clustering of generators/loads so all 8760 hours are simulated in genx

rebeccamccabe committed:
uncomment conda activation in run_all.sh - this might break ci but is needed for running manually

rebeccamccabe committed:
just clarify comment in run_powergenome.sh

rebeccamccabe committed:
only run one wave cost, power limit, zeta, omega_n

kp-579 committed:
used run_genx_case_simple to get marginal thermal generators

MadisonDietrich committed:
get rid of the repmat stuff for the 480x1 data

Commit:
changed test_settings.yml

Commit:
Resolve merge conflicts on cem-ci (accept upstream notebooks)

Commit:
Merge branch 'cem-ci' into run_genx_case_simple

Commit:
ci_cem.yml fixed julia version

Commit:
Trigger CI

Commit:
calkit_run.yml - changed run --versbose to run-action@v2

Commit:
calkit run issue

rj426-cpu committed:
Update ci_cem.yml

rj426-cpu committed:
Update calkit_run.yml

Commit:
Update calkit_run.yml with proper action parameters

Commit:
Merge branch 'cem-ci' of https://github.com/symbiotic-engineering/WEC-DECIDER into cem-ci

Commit:
Update Julia 1.9 kernel

rebeccamccabe committed:
Merge branch 'main' into cem-ci

MadisonDietrich committed:
Merge branch 'cem-ci' of github.com:symbiotic-engineering/WEC-DECIDER into cem-ci

rj426-cpu committed:
Update run_all.sh

rj426-cpu committed:
Update calkit_run.yml

rj426-cpu committed:
Update ci_cem.yml

rj426-cpu committed:
Update ci_cem.yml

rj426-cpu committed:
Update calkit_run.yml

rj426-cpu committed:
Update calkit_run.yml

rj426-cpu committed:
Update ci_cem.yml

rj426-cpu committed:
Update calkit_run.yml

petebachant commented:
Looks like there's a bunch of iteration happening on the automation here. Anything I can help with?

rj426-cpu committed:
Update calkit_run.yml

rj426-cpu committed:
Update ci_cem.yml

rj426-cpu committed:
Update run_all.sh

rebeccamccabe commented:
@rj426-cpu if you are running into any issues with calkit, @petebachant is a great help!
Please be careful with the `sudo rm -rf` commands. I don't think these should be necessary, what issue are you running into?

rj426-cpu committed:
Update ci_cem.yml

rj426-cpu commented:
It gives the error that ERROR: database or disk is full and it basically gave the warning that WARNING: Some of the cache files do not exist neither locally nor on remote.

rj426-cpu committed:
Update calkit_run.yml with removed disk space

rebeccamccabe commented:
Ok, I suspect that is the dvc cache being full not the CI server itself.

From https://github.com/actions/cache?tab=readme-ov-file#cache-limits,
looks like the limit is 10GB per repo. I think this should be fine since
the run actions runs calkit dvc pull before the pipeline, it might just
take longer.

-Becca


On Thu, Dec 11, 2025 at 11:54 AM rj426-cpu ***@***.***> wrote:

> *rj426-cpu* left a comment (symbiotic-engineering/WEC-DECIDER#8)
> <https://github.com/symbiotic-engineering/WEC-DECIDER/pull/8#issuecomment-3642840557>
>
> It gives the error that ERROR: database or disk is full and it basically
> gave the warning that WARNING: Some of the cache files do not exist neither
> locally nor on remote.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/symbiotic-engineering/WEC-DECIDER/pull/8#issuecomment-3642840557>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHUJPSVWND5XNJNDFICLIWD4BGOU7AVCNFSM6AAAAACC2WI456VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTMNBSHA2DANJVG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

petebachant commented:
You could try clearing the GitHub Actions cache: https://github.com/symbiotic-engineering/WEC-DECIDER/actions/caches

It might have a bunch of old artifacts in there from previous runs.

More generally, what is the goal for each of the Actions workflows? `ci_cem` does some environment setup then calls a `run_all.sh` script, which appears to serve the same purpose as the Calkit pipeline, but with no output caching/skipping. Maybe these could be merged into one workflow, unless the goal for `ci_cem` is testing the code, and the Calkit workflow is for efficiently generating artifacts like figures or datasets (skipping ones that are up-to-date).

rebeccamccabe commented:
The goal is for `ci_cem` to eventually be replaced by the calkit pipeline,
and is there to serve as a reference as we are getting the pipeline to work
as expected.


On Thu, Dec 11, 2025 at 1:19 PM Pete Bachant ***@***.***>
wrote:

> *petebachant* left a comment (symbiotic-engineering/WEC-DECIDER#8)
> <https://github.com/symbiotic-engineering/WEC-DECIDER/pull/8#issuecomment-3643209438>
>
> You could try clearing the GitHub Actions cache:
> https://github.com/symbiotic-engineering/WEC-DECIDER/actions/caches
>
> It might have a bunch of old artifacts in there from previous runs.
>
> More generally, what is the goal for each of the Actions workflows? ci_cem
> does some environment setup then calls a run_all.sh script, which appears
> to serve the same purpose as the Calkit pipeline, but with no output
> caching/skipping. Maybe these could be merged into one workflow, unless the
> goal for ci_cem is testing the code, and the Calkit workflow is for
> efficiently generating artifacts like figures or datasets (skipping ones
> that are up-to-date).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/symbiotic-engineering/WEC-DECIDER/pull/8#issuecomment-3643209438>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHUJPSX54M2YGRO35X2FCVD4BGYTJAVCNFSM6AAAAACC2WI456VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTMNBTGIYDSNBTHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

MadisonDietrich committed:
clean up the code

MadisonDietrich committed:
Merge branch 'cem-ci' of github.com:symbiotic-engineering/WEC-DECIDER into cem-ci
